src/greenlight/runner.clj

```diff
@@ -66,32 +56,54 @@
           (map (fn [v] (v)))))))
 
 
+(defn filter-test-suite
+  [test-suite arguments]
+  (if-let [is-only? (= ":only" (first arguments))]
+    (filter
+      (comp (partial re-matches (re-pattern (second arguments))) name ::test/ns)
+      (find-tests))
```

Comment:
It's a little strange to me that this could reach outside of the `test-suite` passed in to find tests.

Reply:
I can restrict filter to tests passed, no problem. But I wonder if we should treat *-test.jar as a collection of tests and by default `lein run test` should run all tests defined in the project and `lein run test :only` used to restrict the test suite. Or even better `lein greenlight` to run all tests and `lein greenlight :only namespace` to run a particular set of tests. But you are right, that it should be a different change focused on that change in the pattern. In this case, I will update the PR to filter the test suite passed in the next commit.